lib/engine/game/g_1860.rb

```diff
@@ -450,6 +528,56 @@ def compute_distance_graph(corporation)
         @hex_distances[corporation] = h_distances
       end
 
+      # needed for custom_node_walk
+      def custom_node_select(node, paths, corporation: nil)
+        on = paths.map { |p| [p, 0] }.to_h
+
+        custom_node_walk(node, on: on, corporation: corporation) do |path|
+          on[path] = 1 if on[path]
+        end
+
+        val = on.keys.select { |p| on[p] == 1 }
+        val
+      end
+
+      # needed for custom_blocks?
+      def custom_node_walk(node, visited: nil, on: nil, corporation: nil, visited_paths: {})
```

Comment:
what's this do?

Reply:
Fundamentally I need a version of Node::select that uses a custom City::blocks? method that understands about flipped tokens for bankrupt corporation.

Reply:
I thought about temporarily replacing flipped tokens with neutral tokens during route selection and dividend steps, but that seemed too inelegant and prone to bugs.